src/Extractor/LinkCollectionExtractorInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+/**
+ * @license   http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ * @copyright Copyright (c) 2014 Zend Technologies USA Inc. (http://www.zend.com)
+ */
+
+namespace ZF\Hal\Extractor;
+
+use Zend\Stdlib\Extractor\ExtractionInterface;
+
+interface LinkCollectionExtractorInterface extends ExtractionInterface
+{
+    /**
+     * @return LinkExtractorInterface
+     */
+    public function getLinkExtractor();
+
+    /**
+     * @param LinkExtractorInterface $linkExtractor
+     */
+    public function setLinkExtractor(LinkExtractorInterface $linkExtractor);
```

Comment:
Do we need to define this? or can we leave that up to the implementation to determine how a LinkExtractor is injected into the LinkCollectionExtractor? Trying to avoid setters as much as possible…